enumerables.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    elements.size.times do
+      yield(elements[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    size.times do
+      yield(elements[index], index)
+      index += 1
+    end
+  end
+
+  def my_select(&array)
+    return enum_for unless block_given?
+
+    selected_items = []
+    my_each do |x|
+      selected_items.push(x) unless array.call(x) == false
+    end
+    selected_items
+  end
+
+  def my_all?(*array)
+    array = nil
+    if array
+      my_each { |x| return false unless array == x }
+    elsif !block_given?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return false unless yield(x) }
+    end
+    true
+  end
+
+  def my_any?(*args)
+    args = nil
+    if args
+      my_each { |x| return true if x == args }
+    end
+    if !block_given?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return false unless x == yield(x) }
+    end
+    false
+  end
+
+  def my_none?(*args, &block)
+    !my_any?
+  end
+
+  def my_count(arguement = nil)
+    counter = 0
+    if arguement
+      my_each { |x| counter += 1 if x == arguement }
+    elsif block_given?
+      my_each { |x| counter += 1 if yield(x) }
+    else
+      counter = length
+    end
+    counter
+  end
+
+  def my_map(proc = nil)
+    return to_enum unless block_given?
+
+    selected = []
+    if proc
+      my_each { |x| selected.push(proc.call(x)) }
+    else
+      my_each { |x| selected.push(yield(x)) }
+    end
+    selected
+  end
+
+  def my_inject(memo = nil, sym = nil, &prc)
+    memo = memo.to_sym if memo.is_a?(String) && !sym && !prc
+
+    if memo.is_a?(Symbol) && !sym
+      prc = memo.to_proc
+      memo = nil
+    end
+
+    sym = sym.to_sym if sym.is_a?(String)
+    prc = sym.to_proc if sym.is_a?(Symbol)
+
+    my_each { |elem| memo = memo.nil? ? elem : prc.yield(memo, elem) }
+    memo
+  end
+end
+
+  def multiply_els(*item)
+    arrays = Array
+    array = arrays(item)
+    array.my_inject { |item1, item2| item1 * item2 }
+  end
+
+array = [false, true, false, nil, []]
+    puts array.my_any? # should print true
+    #puts array.my_all? # should print false
+  #puts array.my_none? # should print false
+  #array = [7,8,9,5,1,0,2,4,0,1]
+p array.my_count(0) == array.count(0) # false
+words = %w[dog door rod blade]
+  # puts words.my_all?(/c/) == words.all?(/c/) # false
+  # puts words.my_any?(/c/) == words.all?(/c/) # false
+  # puts words.my_none?(/c/) == words.all?(/c/) # true
+  array = [1,7,8,89,9,6,5,12,4,2,5]
+  #p array.my_inject(:+) == array.inject(:+) 
```

Comment:
No debugging instructions should be in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)

Reply:
okay are you on slack